core/dbt/clients/registry.py

```diff
@@ -28,11 +28,14 @@ def wrapper(*args, **kwargs):
             attempt += 1
             try:
                 return fn(*args, **kwargs)
+            except requests.exceptions.Timeout as exc_timeout:
+                raise requests.exceptions.Timeout(
+                    'Request timeout, server has not issued a response for 60 seconds'
+                ) from exc_timeout
```

Comment:
Should this raise its own `Timeout`-specific error message after hanging for 60 seconds, or simply kick back to retry in the line below?

```python
except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
```
I think either could be a good answer!

Reply:
My initial thought was to have 2 distinct error handling, but since, it appears that, the azure pipeline does not raise a `Timeout`error, it might be safer to handle both in a generic error handling block as you wrote above.